editor/src/templates/editor-canvas.tsx

```diff
@@ -1034,54 +1040,36 @@ export class EditorCanvas extends React.Component<EditorCanvasProps> {
                   }),
                 ])
               }
-              const image = images[0]
-
-              const { frame } = getFrameAndMultiplierWithResize(
-                mousePosition.canvasPositionRounded,
-                image.filename,
-                image.size,
-                this.props.editor.canvas.scale,
-              )
 
-              const { saveImageActions, src } = isLoggedIn(this.props.userState.loginState)
-                ? {
-                    saveImageActions: [
-                      EditorActions.saveAsset(
-                        image.filename,
-                        image.fileType,
-                        image.base64Bytes,
-                        image.hash,
-                        EditorActions.saveImageDetails(
-                          image.size,
-                          EditorActions.saveImageReplace(),
-                        ),
-                      ),
-                    ],
-                    src: imagePathURL(image.filename),
-                  }
-                : { saveImageActions: [], src: image.base64Bytes }
-
-              const newUID = generateUidWithExistingComponents(this.props.editor.projectContents)
-              const elementSize: Size = resize(
-                size(frame.width ?? 100, frame.height ?? 100),
-                size(200, 200),
-                'keep-aspect-ratio',
-              )
-              const newElement = createJsxImage(newUID, {
-                width: elementSize.width,
-                height: elementSize.height,
-                top: mousePosition.canvasPositionRounded.y,
-                left: mousePosition.canvasPositionRounded.x,
-                src: src,
-              })
+              let actions: Array<EditorAction> = []
+              let uidsSoFar: Array<string> = []
+              let subjects: Array<ElementInsertionSubject> = []
+              for (const image of images) {
+                const { actions: actionsForImage, singleSubject } = actionsForDroppedImage(image, {
+                  generateUid: () =>
+                    generateUidWithExistingComponentsAndExtraUids(
+                      this.props.editor.projectContents,
+                      uidsSoFar,
```

Comment:
wouldn't it be better to move this whole for loop into a dedicated function, and just use the return values here?

Reply:
move to a dedicated function